### DISCUSSION TITLE
Setting up Polypheny-DB for development

### DISCUSSION THREAD

**Discussion by S-Harshit** [polypheny/Polypheny-DB#286] [Q&A] [answered]
Hello everyone, I am Harshit I wanted to contribute to Polypheny-DB,I checked the documentation and there is documentation for Polypheny-UI development but I was not able to find any clear instruction on how to set up development environment for Polypheny-DB specifically, how to proceed further after forking the repo or how can I can set it up in an IDE. 
Thank you.

**Comment by vogti** [comment, marked as answer]
Hi S-Harshit, 

you're right, there is not yet a guide for setting up a development environment. I will take care of this. For the time being, here is a short description. The setup is quite easy:

1. Fork and checkout Polypheny-DB
2. Import the project in your favorite IDE as gradle project. Most of us are using IntelliJ. But other IDEs should work fine as well.
3. Set up lombok support in your IDE. See [this tutorial](https://www.baeldung.com/lombok-ide) for further information.
4. Import our coding style (I have attached the latest version to this post. There has been a minor change to the one in the Admin Repo).
5. Add the copyright profile (either configure manually with the header on the website or -- if you are using IntelliJ -- copy the content of the attached zip file to the `.idea/copyright` folder in the Polypheny-DB repo).
6. Start Polypheny-DB using its main-Class (`dbms/src/main/java/org/polypheny/db/PolyphenyDb.java`). There are no arguments required. 

If you would like to reset the catalog (and thus Polypheny) you can use the argument `-resetCatalog` on startup.

There is no need to manually set up the UI if you do not plan to work at the UI. Polypheny-DB pulls the latest version of the UI in the build process.

Please give me feedback if I have missed something.

[Polypheny-DB Style (extends GoogleStyle).xml.zip](https://github.com/polypheny/Polypheny-DB/files/6117373/Polypheny-DB.Style.extends.GoogleStyle.xml.zip) (As zip since GitHub does not allow uploading .xml files)
[copyright.zip](https://github.com/polypheny/Polypheny-DB/files/6117374/copyright.zip)

**Reply by S-Harshit** [reply]
@vogti I want Polypheny-DB to use the Polypheny-UI that I forked and set up for development at port 4200 according to Polypheny-UI documentation, is it possible?

**Reply by vogti** [reply]
What do you mean with "use"? If you start Polypheny-UI according to the documentation you can use your modified version running on port 4200 to interact with Polypheny-DB. But on port 8080 it still provides you with the latest master version of Polypheny-UI. Do you want to replace this one with your own version? This is indeed possible: First you need to do a `./gradlew clean install` in your fork of the Polypheny-UI, and then you do a `./gradlew :webui:clean :webui:build` in Polypheny-DB. If you now restart Polypheny-DB, it will provide your modified version of Polypheny-UI on port 8080. But be aware of the following disadvantage: Since your version of Polypheny-UI is now installed into the local maven repository on your machine, it will always override the version in the remote maven repository (at least as long as we do not increase the version number). This can easily be undone by deleting the `.m2/repository/org/polypheny/polypheny-ui` directory in your home folder.
